tracer/src/Datadog.Trace/Activity/OtlpHelpers.cs

```diff
@@ -387,27 +391,55 @@ private static void ExtractExceptionAttributes<TInner>(TInner activity, Span spa
                     continue;
                 }
 
+                string? errorType = null;
+                string? errorMsg = null;
+                string? errorStack = null;
+
                 foreach (var tag in duckEvent.Tags)
                 {
                     switch (tag.Key)
                     {
                         case OpenTelemetryErrorType:
-                            SetTagObject(span, Tags.ErrorType, tag.Value);
+                            errorType = tag.Value?.ToString();
                             break;
 
                         case OpenTelemetryErrorMsg:
-                            SetTagObject(span, Tags.ErrorMsg, tag.Value);
+                            errorMsg = tag.Value?.ToString();
                             break;
 
                         case OpenTelemetryErrorStack:
-                            SetTagObject(span, Tags.ErrorStack, tag.Value);
+                            errorStack = tag.Value?.ToString();
                             break;
                     }
                 }
 
-                // we've found the exception attribute so we should be done here
+                // Ensure that all of the error tracking tags are updated (even null values) from the same exception attribute
```

Comment:
How is setting the tags to null different from not setting these fields at all? Do we need this behavior to support the error tracking product?

Reply:
Setting the tag to null is equivalent to unsetting the field, which is why on each iteration I have to set all of them. If, for example, there was not an error message, we'll unset the previous error message by setting the input value as `null`